File: app/src/main/java/com/lidchanin/shoppinglist/activities/TutorialActivity.java
package com.lidchanin.shoppinglist.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

import com.lidchanin.shoppinglist.R;
import com.lidchanin.shoppinglist.adapters.TutorialPagerAdapter;
import com.lidchanin.shoppinglist.customview.DesignedViewPager;
import com.lidchanin.shoppinglist.utils.TutorialManager;

import java.util.List;


public class TutorialActivity extends AppCompatActivity {

    private DesignedViewPager designedViewPager;
    private TabLayout tabLayout;
    private Button previous;
    private Button next;

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }
        setContentView(R.layout.activity_main);
        designedViewPager = (DesignedViewPager) findViewById(R.id.viewPager);
        tabLayout = (TabLayout) findViewById(R.id.tabLayout);
        final List<Integer> tutorialImages = TutorialManager.getTutorialImageList(this);
        List<String> tutorialStrings = TutorialManager.getTutorialTextList(this);
        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabLayout);
        tabLayout.setupWithViewPager(designedViewPager, true);
        previous = (Button) findViewById(R.id.previous);
        previous.setVisibility(View.GONE);
        previous.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(designedViewPager.getCurrentItem()-1>=0) {
                    designedViewPager.setCurrentItem(designedViewPager.getCurrentItem()-1);
                }
                if(designedViewPager.getCurrentItem() == 0 ){
                    previous.setVisibility(View.GONE);
                }
            }
        });
        next = (Button) findViewById(R.id.next);
        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                previous.setVisibility(View.VISIBLE);
                if(designedViewPager.getCurrentItem()+1<tutorialImages.size()) {
                    designedViewPager.setCurrentItem(designedViewPager.getCurrentItem()+1);
                }else {
                    // TODO: 03.12.2017 make animation to hide load
                    startActivity(new Intent(TutorialActivity.this, ShoppingListFragmentManager.class));
                }
            }
        });
        designedViewPager.setOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                if(designedViewPager.getCurrentItem() == 0 ){
                    previous.setVisibility(View.GONE);
                }else {
                    previous.setVisibility(View.VISIBLE);
                }
            }

            @Override
            public void onPageSelected(int position) {

            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        designedViewPager.setAdapter(new TutorialPagerAdapter(getSupportFragmentManager(), tutorialImages, tutorialStrings));
    }



}

File: app/src/main/java/com/lidchanin/shoppinglist/activities/TutorialActivity.kt
package com.lidchanin.shoppinglist.activities

import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.lidchanin.shoppinglist.R
import com.lidchanin.shoppinglist.adapters.TutorialPagerAdapter
import com.lidchanin.shoppinglist.utils.TutorialManager
import kotlinx.android.synthetic.main.activity_main.*

class TutorialActivity : AppCompatActivity(){


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (supportActionBar != null) {
            supportActionBar!!.hide()
        }
        setContentView(R.layout.activity_main)
        val tutorialImages = TutorialManager.getTutorialImageList(this)
        val tutorialStrings = TutorialManager.getTutorialTextList(this)
        tabLayout.setupWithViewPager(viewPager, true)
        previous.visibility = View.GONE
        previous.setOnClickListener {
            if (viewPager!!.currentItem - 1 >= 0) {
                viewPager!!.currentItem = viewPager!!.currentItem - 1
            }
            if (viewPager!!.currentItem == 0) {
                previous!!.visibility = View.GONE
            }
        }
        next!!.setOnClickListener {
            previous!!.visibility = View.VISIBLE
            if (viewPager!!.currentItem + 1 < tutorialImages.size) {
                viewPager!!.currentItem = viewPager!!.currentItem + 1
            } else {
                startActivity(Intent(this@TutorialActivity, ShoppingListFragmentManager::class.java))
            }
        }
        viewPager!!.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
                if (viewPager!!.currentItem == 0) {
                    previous!!.visibility = View.GONE
                } else {
                    previous!!.visibility = View.VISIBLE
                }
            }

            override fun onPageSelected(position: Int) {

            }

            override fun onPageScrollStateChanged(state: Int) {

            }
        })
        viewPager!!.adapter = TutorialPagerAdapter(supportFragmentManager, tutorialImages, tutorialStrings)
    }

}